src/ruis/render/scene/gltf_loader.cpp

```diff
@@ -0,0 +1,772 @@
+/*
+carcockpit - Car cockpit example GUI project
+
+Copyright (C) 2024 Gagistech Oy <gagisechoy@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+/* ================ LICENSE END ================ */
+
+#include "gltf_loader.hxx"
+
+#include <jsondom/dom.hpp>
+#include <papki/span_file.hpp>
+#include <rasterimage/image_variant.hpp>
+#include <utki/deserializer.hpp>
+#include <utki/string.hpp>
+#include <utki/util.hpp>
+
+using namespace std::string_view_literals;
+using namespace ruis::render;
+
+accessor::accessor(
+	utki::shared_ref<buffer_view> bv,
+	uint32_t count,
+	uint32_t byte_offset,
+	type type_v,
+	component_type component_type_v
+) :
+	bv(std::move(bv)),
+	count(count),
+	byte_offset(byte_offset),
+	type_v(type_v),
+	component_type_v(component_type_v)
+{}
+
+gltf_loader::gltf_loader(ruis::render::factory& factory_v) :
+	factory_v(factory_v)
+{}
+
+namespace {
+
+int read_int(const jsondom::value& json, const std::string& name, const int default_value = -1)
+{
+	auto it = json.object().find(name);
+	if (it != json.object().end())
+		return it->second.number().to_int32();
+
+	return default_value;
+}
+
+uint32_t read_uint(const jsondom::value& json, const std::string& name, const uint32_t default_value = 0)
+{
+	auto it = json.object().find(name);
+	if (it != json.object().end())
+		return it->second.number().to_uint32();
+
+	return default_value;
+}
+
+// float read_float(const jsondom::value& json, const std::string& name, const float default_value = 0.0f)
```

Comment:
remove commented code. It is not so difficult to create it again when needed by copy-pasting from `read_uint()`.

Reply:
removed